carla/evaluation/benchmark.py

```diff
@@ -0,0 +1,114 @@
+import csv
+from typing import Dict
+
+import pandas as pd
+
+from carla.evaluation.distances import get_distances
+from carla.models.api import MLModel
+from carla.models.pipelining import encode, scale
+from carla.recourse_methods.api import RecourseMethod
+
+
+class Benchmark:
+    def __init__(
+        self, mlmodel: MLModel, recmodel: RecourseMethod, factuals: pd.DataFrame
+    ) -> None:
+        """
+        Constructor for benchmarking class
+
+        Parameters
+        ----------
+        mlmodel: MLModel
+            Black Box model we want to explain
+        recmodel: RecourseMethod
+            Recourse method we want to benchmark
+        factuals: pd.DataFrame
+            Instances we want to find counterfactuals
+        """
+        self._recmodel = recmodel
+        self._counterfactuals = self._recmodel.get_counterfactuals(factuals)
+
+        # Normalizing and encoding factual for later use
+        self._factuals = scale(mlmodel.scaler, mlmodel.data.continous, factuals)
+        self._factuals = encode(
+            mlmodel.encoder, mlmodel.data.categoricals, self._factuals
+        )
+        self._factuals = self._factuals[
+            mlmodel.feature_input_order + [mlmodel.data.target]
+        ]
+
+    def compute_distances(self) -> Dict:
+        """
+        Calculates the distance measure and returns it as dict
+
+        Returns
+        -------
+        Dict
+        """
+        key = "Distances"
+        output: Dict = {key: []}
+        arr_f = self._factuals.to_numpy()
```

Comment:
Why do we need to convert to numpy here? Would it make sense to make the distance functions also accept the current factual type?

Reply:
I think we agreed to only accept arrays for measurements, while our factuals and counterfactuals are DataFrames.

Reply:
Ok, if we agreed to that i am fine 👍🏽